src/ripple/rpc/handlers/ValidatorInfo.cpp

```diff
@@ -30,24 +30,28 @@ Json::Value
 doValidatorInfo(RPC::JsonContext& context)
 {
     // return error if not configured as validator
-    if (context.app.getValidationPublicKey().empty())
+    auto const validationPK = context.app.getValidationPublicKey();
+    if (!validationPK)
         return RPC::not_validator_error();
 
     Json::Value ret;
 
-    auto const pk = context.app.getValidationPublicKey();
-
-    // assume pk is ephemeral key, get master key
-    auto const mk = context.app.validatorManifests().getMasterKey(pk);
+    // assume validationPK is ephemeral key, get master key
+    auto const mk =
+        context.app.validatorManifests().getMasterKey(*validationPK);
     ret[jss::master_key] = toBase58(TokenType::NodePublic, mk);
 
-    // pk is maskter key, eg no ephemeral key, eg no manifest, just return
-    if (mk == pk)
+    // validationPK is master key, this implies that there is no ephemeral
+    // key, no manifest, hence return
+    if (mk == validationPK)
         return ret;
 
     // lookup ephemeral key
     auto const ek = context.app.validatorManifests().getSigningKey(mk);
-    ret[jss::ephemeral_key] = toBase58(TokenType::NodePublic, ek);
+
+    assert(ek == validationPK);
```

Comment:
thanks for this observation.
I was mimicking the structure of the current code base and didn't see the larger picture. I have removed this piece of code.

There is a significant overlap in the information stored between `validatorManifests`, `ValidatorKeys` and `ValidatorList`. It would be very helpful if I could refactor these classes to use a single source of truth, instead of duplicate values smattered around.

but that would be a battle for another PR